curve_apps/trend_lines/driver.py

```diff
@@ -108,12 +108,14 @@ def get_connections(self) -> tuple:
             if len(ind) < 2:
                 continue
 
+            detection_params = dict(self.params.detection)
+            if detection_params["max_distance"] is None:
+                detection_params["max_distance"] = np.inf
```

Comment:
To go along with my comment on geoapps-utils, could this work?:
```
segments = find_curves(
    self.vertices[ind, :2],
    self.parts[ind],
    self.params.detection
)
```
You'd obviously have to move some of your logic over to the opts class..

Reply:
yep, moved it over.